examples/src/main/java/kafka/examples/ExactlyOnceMessageProcessor.java

```diff
@@ -155,13 +151,20 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
                     // Finish the transaction. All sent records should be visible for consumption now.
                     producer.commitTransaction();
                     messageProcessed += records.count();
```

Comment:
I had a question before about the "groupMode." Do we need to include this in the example? I think it would be fine to let this example use the latest recommended pattern and include a comment about it.

Also, could we have a helper for the boilerplate conversion to `Map<TopicPartition, OffsetAndMetadata`>` since it clutters up the core logic?